daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -396,11 +396,8 @@ private[lf] final class Compiler(
       case ECons(_, front, tail) =>
         // TODO(JM): Consider emitting SEValue(SList(...)) for
         // constant lists?
-        val args = (front.iterator.map(compile) ++ Seq(compile(tail))).toArray
-        if (front.length == 1) {
-          SEApp(SEBuiltin(SBCons), args)
-        } else {
-          SEApp(SEBuiltin(SBConsMany(front.length)), args)
+        front.foldRight(compile(tail)) { (head, acc) =>
```

Comment:
Is it fine to compile the argument in reverse order ?

Reply:
Yes, as long as they are executed in the right order, which they seem to be, everything is alright. If it wasn't, the following test would fail:
https://github.com/digital-asset/daml/blob/0ee2159218eda085a084ed07ca2d7b245fbb6c8e/compiler/damlc/tests/daml-test-files/List.daml#L51